### ISSUE TITLE
when i try to train, an error happend

### ISSUE THREAD

**Issue by cafebaby** [Stick-To/RefineDet-tensorflow#2]
Caused by op 'predictor/while/ArgMax_1', defined at:
  File "/Users/PycharmProjects/2019/RefineDet-tensorflow/test.py", line 78, in <module>
    refinedet = net.RefineDet320(config, trainset_provider)
  File "/Users/PycharmProjects/2019/RefineDet-tensorflow/RefineDet320.py", line 53, in __init__
    self._build_graph()
  File "/Users/PycharmProjects/2019/RefineDet-tensorflow/RefineDet320.py", line 195, in _build_graph
    state = tf.while_loop(cond, body, init_state)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 3291, in while_loop
    return_same_structure)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 3004, in BuildLoop
    pred, body, original_loop_vars, loop_vars, shape_invariants)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2939, in _BuildLoop
    body_result = body(*packed_vars_for_body)
  File "/Users/PycharmProjects/2019/RefineDet-tensorflow/RefineDet320.py", line 189, in <lambda>
    tf.squeeze(tf.gather(self.ground_truth, tf.cast(i, tf.int32))),
  File "/Users/PycharmProjects/2019/RefineDet-tensorflow/RefineDet320.py", line 593, in _compute_one_image_loss
    rgindex = tf.argmax(other_agiou_rate, axis=1)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py", line 88, in argmax
    return gen_math_ops.arg_max(input, axis, name=name, output_type=output_type)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/ops/gen_math_ops.py", line 787, in arg_max
    name=name)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3274, in create_op
    op_def=op_def)
  File "/Users/venv/machine/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1770, in __init__
    self._traceback = tf_stack.extract_stack()

InvalidArgumentError (see above for traceback): Reduction axis 1 is empty in shape [6375,0]
   [[node predictor/while/ArgMax_1 (defined at /Users/PycharmProjects/2019/RefineDet-tensorflow/RefineDet320.py:593)  = ArgMax[T=DT_FLOAT, Tidx=DT_INT32, output_type=DT_INT64, _device="/job:localhost/replica:0/task:0/device:CPU:0"](predictor/while/boolean_mask_8/GatherV2, predictor/while/ArgMax_1/dimension)]]

Process finished with exit code 1

**Comment by nguoer121** [comment]
the problem is data augmentation: zoom and crop, this operation affects the labeled box at the edge, you can remove the variables('zoom_size': [340, 340],  'crop_method': 'random',) of  "image_augmentor_config".

**Comment by cafebaby** [comment]
> the problem is data augmentation: zoom and crop, this operation affects the labeled box at the edge, you can remove the variables('zoom_size': [340, 340], 'crop_method': 'random',) of "image_augmentor_config".

thank you so mush, I remove the variables and It's working pretty well right now

**Comment by Stick-To** [comment]
If  a bbox center (ycenter, center) is not in image after image augmentation, this piece of bbox would be removed. If all pieces of bbox were removed, this problem will raise.  so you should select appropriate image augmentation strategy.
